functionary/core/tests/utils/test_parameter.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+from django.core.exceptions import ValidationError
+
+from core.models import Function, FunctionParameter, Package, Task, Team
+from core.utils.parameter import PARAMETER_TYPE, validate_parameters
+
+
+@pytest.fixture
+def team():
+    return Team.objects.create(name="team")
+
+
+@pytest.fixture
+def environment(team):
+    return team.environments.get()
+
+
+@pytest.fixture
+def package(environment):
+    return Package.objects.create(name="testpackage", environment=environment)
+
+
+@pytest.fixture
+def function(package):
+    return Function.objects.create(
+        name="testfunction",
+        package=package,
+        environment=package.environment,
+    )
+
+
+@pytest.fixture
+def param1(function):
+    return FunctionParameter.objects.create(
+        name="json_param",
+        function=function,
+        parameter_type=PARAMETER_TYPE.JSON,
+        required=True,
+    )
+
+
+@pytest.fixture
+def param2(function):
+    return FunctionParameter.objects.create(
+        name="dont_hide",
+        function=function,
+        parameter_type=PARAMETER_TYPE.DATE,
+        required=True,
+    )
+
+
+@pytest.fixture
+def task(function, environment, admin_user):
+    return Task.objects.create(
+        function=function,
+        environment=environment,
+        parameters={},
+        creator=admin_user,
+    )
+
+
+@pytest.mark.django_db
+@pytest.mark.usefixtures("param1")
+def test_parameters(task):
+    """JSON parameters get stringified."""
+    validate_parameters({"json_param": {"hello": 1}}, task.function)
+
+
+@pytest.mark.django_db
+def test_functions_without_parameters(task):
+    """Check a parameterless function passes."""
+    validate_parameters({}, task.function)
+    validate_parameters({"json_param": {"hello": 1}}, task.function)
+
+
+@pytest.mark.django_db
+@pytest.mark.usefixtures("param1", "param2")
+def test_missing_parameter(task):
+    """Check for missing param2"""
+    with pytest.raises(ValidationError, match=r".*dont_hide.*"):
+        validate_parameters({"json_param": {"hello": 1}}, task.function)
+
+
+@pytest.mark.django_db
+@pytest.mark.usefixtures("param1")
+def test_incorrect_parameters(task):
+    """Check that incorrect parameters don't work."""
+    with pytest.raises(ValidationError, match=r".*json_param.*is a required.*"):
+        validate_parameters({}, task.function)
+
+    with pytest.raises(ValidationError, match=r".*json_param.*is a required.*"):
+        validate_parameters({"true": False}, task.function)
+
+
+@pytest.mark.django_db
+@pytest.mark.usefixtures("param1", "param2")
+def test_incorrect_parameters2(task):
```

Comment:
A slight tweak to the naming and usage of the fixtures could make these tests easier to read and understand in isolation.  For example:

```python
@pytest.mark.django_db
def test_incorrect_parameter_type(task, json_param, date_param):
    """Check that parameters with an incorrect type don't work."""
    with pytest.raises(ValidationError, match=r".*date_param.*"):
        validate_parameters({json_param.name: 1, date_param.name: False}, task.function)

    with pytest.raises(ValidationError, match=r".*date_param.*"):
        validate_parameters(
            {json_param.name: 1, date_param.name: "False"}, task.function
        )
```

I feel like it's more obvious reading the above that `date_param` is being provided as the incorrect type, as where with `dont_hide` I have to go look at the fixture.

One other minor note is that it doesn't look like you're using the task fixture itself anywhere.  You could just use the function fixture directly and do away with the task.